internal/api/mfa.go

```diff
@@ -71,14 +147,21 @@ func (a *API) EnrollFactor(w http.ResponseWriter, r *http.Request) error {
 	if session == nil || user == nil {
 		return internalServerError("A valid session and a registered user are required to enroll a factor")
 	}
-
 	params := &EnrollFactorParams{}
 	if err := retrieveRequestParams(r, params); err != nil {
 		return err
 	}
 
-	if params.FactorType != models.TOTP {
-		return badRequestError(ErrorCodeValidationFailed, "factor_type needs to be totp")
+	if params.FactorType == models.SMS {
+		return a.enrollSMSFactor(w, r, user, session, params)
+	}
+	if params.FactorType != models.TOTP && params.FactorType != models.SMS {
+		return badRequestError(ErrorCodeValidationFailed, "factor_type needs to be totp or sms")
+	}
+	if params.FactorType == models.TOTP {
+		if !config.MFA.TOTP.EnrollEnabled {
+			return unprocessableEntityError(ErrorCodeMFATOTPEnrollDisabled, "MFA enroll is not enabled for TOTP Factors")
+		}
```

Comment:
nit but i think this is a cleaner approach - we can check the `EnrollEnabled` here, currently, this is done here for TOTP and in the `enrollSMSFactor` method

```suggestion
        switch params.FactorType {
	case models.SMS:
		if !config.MFA.SMS.EnrollEnabled {
			return unprocessableEntityError(ErrorCodeMFASMSEnrollDisabled, "MFA enroll is disabled for SMS")
		}
		return a.enrollSMSFactor(w, r, user, session, params)
	case models.TOTP:
		if !config.MFA.TOTP.EnrollEnabled {
			return unprocessableEntityError(ErrorCodeMFATOTPEnrollDisabled, "MFA enroll is not enabled for TOTP")
		}
	default:
		return badRequestError(ErrorCodeValidationFailed, "factor_type needs to be totp or sms")
	}
```

Reply:
Yup, let's do that